src/Analyzer/ArrayJoinNode.cpp

```diff
@@ -72,7 +72,8 @@ ASTPtr ArrayJoinNode::toASTImpl(const ConvertToASTOptions & options) const
         else
             array_join_expression_ast = array_join_expression->toAST(options);
 
-        array_join_expression_ast->setAlias(array_join_expression->getAlias());
+        if (array_join_expression->hasAlias())
```

Comment:
```suggestion
        /// We must check that it has an alias (not empty) as otherwise we try to set it and not all IAST classes support it (LOGICAL_ERROR)
        if (array_join_expression->hasAlias())
```